src/operator/roi_pooling.cc

```diff
@@ -74,7 +79,13 @@ inline void ROIPoolForward(const Tensor<cpu, 4, Dtype> &out,
 
     const Dtype* batch_data = bottom_data + data_size * roi_batch_ind;
 
+    #pragma omp parallel for firstprivate(batch_data, top_data, argmax_data)
     for (int c = 0; c < channels_; ++c) {
+      // Increment all data pointers
+      const Dtype* batch_data_c = batch_data + c * data_size_c;
+      Dtype* top_data_c = top_data + c * out_size_c;
+      Dtype* argmax_data_c = argmax_data + c * max_idx_size_c;
+
```

Comment:
Nice work!
By the way, Multiplies are a lot slower than adds, so you can get even faster by reducing the muls in the loop ( + and * replace by single +):
```cpp
const Dtype *batch_data_c = batch_data;
Dtype *top_data_c = top_data, *argmax_data_c = argmax_data; 
for (int c = 0; c < channels_; ++c) {
  // Increment all data pointers
  batch_data_c  += data_size_c;
  top_data_c    += out_size_c;
  argmax_data_c += max_idx_size_c;
  .
  .
  .
}
```

Reply:
In other places too...

Reply:
Sorry, I mean like this:
```cpp
const Dtype *batch_data_c = batch_data;
Dtype *top_data_c = top_data, *argmax_data_c = argmax_data;
for (int c = 0; c < channels_; ++c, 
                               batch_data_c  += data_size_c,
                               top_data_c    += out_size_c,
                               argmax_data_c += max_idx_size_c 
) {

}
```

Reply:
@cjolivier01 Thanks for the great comments to Xinyu.
Regarding replacing multiplication (FMA) to incremental addition (+=), two points of my view:

1)  The incremental addition is more concise and logical rather than computing the index from start point each time by multiplication. But there're strong compute dependency and we **CANT** start the loop N+1 before the loop N. 
Thus, we have to change to multiple styles for the parallelization.

2) The efficiency of Multiplication (FMA) and addition (ADD) is same in the latest HW (Intel skylake)
Take SSE instruct as an example:
ADD, latency 4, CPI 0.5;
https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=add&techs=SSE&expand=127,3673,127
FMA (MUL+ADD), latency 4, CPI 0.5
https://software.intel.com/sites/landingpage/IntrinsicsGuide/#expand=127,3673,2395,3673,2395,2407&text=mul&techs=FMA,Other

Reply:
ok

Reply:
try to keep in mind in the future that we support less-awesome CPUs than Intel Skylake :)